pkg/automation/automation.go

```diff
@@ -190,16 +190,39 @@ func installLicenseSecret(clientset *kubernetes.Clientset, licenseSecret corev1.
 		return appSlug, errors.Wrap(err, "failed to create app record")
 	}
 
-	// airgap data is the airgap manifest + app specs + image list laoded from configmaps
-	airgapData, err := getAirgapData(clientset, verifiedLicense)
-	if err != nil {
-		return appSlug, errors.Wrapf(err, "failed to load airgap data for %s", appSlug)
-	}
+	if !kotsadm.IsAirgap() {
+		createAppOpts := online.CreateOnlineAppOpts{
+			PendingApp: &onlinetypes.PendingApp{
+				ID:           a.ID,
+				Slug:         a.Slug,
+				Name:         a.Name,
+				LicenseData:  string(license),
+				VersionLabel: instParams.AppVersionLabel,
+			},
+			UpstreamURI:            upstreamURI,
+			IsAutomated:            true,
+			SkipPreflights:         instParams.SkipPreflights,
+			SkipCompatibilityCheck: instParams.SkipCompatibilityCheck,
+		}
+		_, err = online.CreateAppFromOnline(createAppOpts)
+		if err != nil {
+			return appSlug, errors.Wrap(err, "failed to create online app")
+		}
+	} else {
+		// airgap data is the airgap manifest + app specs + image list laoded from configmaps
+		airgapData, err := getAirgapData(clientset, verifiedLicense)
```

Comment:
why if there's no airgap data but it's an airgapped installation. e.g. user passes `--airgap` without `--airgap-bundle`

Reply:
addressed during refactor of automated install